frontend/src/RunCode.js

```diff
@@ -31,9 +31,10 @@ function initWorker() {
 initWorker();
 
 const channel = makeChannel({serviceWorker: {scope: "/course/"}});
-if (channel?.type === "serviceWorker") {
-  navigator.serviceWorker.register("./service-worker.js");
-}
+// // "also you should probably comment out this code in RunCode.js just to avoid having multiple places that register the service worker:"
+// if (channel?.type === "serviceWorker") {
+//   navigator.serviceWorker.register("./service-worker.js");
+// }
```

Comment:
You can remove this. Do you think you can make the service worker always register, but by default disable the caching on localhost from within the service worker itself?

Reply:
Good thinking. I think it's working this way now.

Reply:
(Feel free to resolve when you've tested it too)